### ISSUE TITLE
NET-1910 Verifier.VerifyCodeFix cannot correctly handle snippets with expressions with more than one issue

### ISSUE THREAD

**Issue by costin-zaharia-sonarsource** [SonarSource/sonar-dotnet#4468]
### Description

`Verifier.VerifyCodeFix` cannot correctly handle snippets with expressions with more than one issue.

### Repro steps

```
		public bool Method1(bool condition)
        {
            return condition == false // Noncompliant FP
                ? true // Noncompliant FP
                : throw new Exception();
        }
```

When the above snippet is analyzed by the `Verifier.VerifyCodeFix`, it will run the code fix in 2 different ways which have a different end result. See https://github.com/SonarSource/sonar-dotnet/blob/1711a5f773013925e079d11e32bf0a3f592a6706/analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/CodeFixVerifier.cs line 52 and 63.

`RunCodeFixWhileDocumentChanges` will expect:
```
		public bool Method1(bool condition)
        {
            return !condition || throw new Exception();
        }
```
and `RunFixAllProvider` is expecting:
```
		public bool Method1(bool condition)
        {
            return !condition ? true // Fixed
                : throw new Exception();
        }
```

**Comment by pavel-mikula-sonarsource** [comment]
I understand this as an intentional test case.
Currently, the code fix execution forces us to provide the final version of the fix. 
Without that, our codefix would provide a code that will contain the same issue again in a little different flavor. Requiring the user to run the code fix for a second time.

I think we should improve the error message to be clear about this scenario, as I find it very difficult to understand - UT fails after making, `fixed` file is updated accordingly, UT fails again 🤯

**Comment by costin-zaharia-sonarsource** [comment]
I see this as a limitation of the verifier since the code example I provided can appear in practice and we want to test how the code-fix provider acts in this case.

**Comment by pavel-mikula-sonarsource** [comment]
Yes, and it actually helped us discover that the codefix doesn't provide what it should. If I understand it correctly, the codefix should provide this:
```
		public bool Method1(bool condition)
        {
            return !condition || throw new Exception();
        }
```
while it was providing something else - something that contained the same problem again.

The pain is that this discovery is very hard to understand.

**Comment by pavel-mikula-sonarsource** [comment]
Moved to [NET-1910](https://sonarsource.atlassian.net/browse/NET-1910)

[NET-1910]: https://sonarsource.atlassian.net/browse/NET-1910?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
